src/validation.cpp

```diff
@@ -620,19 +621,70 @@ static bool AcceptToMemoryPoolWorker(const CChainParams& chainparams, CTxMemPool
             setAncestors.clear();
             // If CalculateMemPoolAncestors fails second time, we want the original error string.
             std::string dummy_err_string;
-            // If the new transaction is relatively small (up to 40k weight)
-            // and has at most one ancestor (ie ancestor limit of 2, including
-            // the new transaction), allow it if its parent has exactly the
-            // descendant limit descendants.
-            //
-            // This allows protocols which rely on distrusting counterparties
-            // being able to broadcast descendants of an unconfirmed transaction
-            // to be secure by simply only having two immediately-spendable
-            // outputs - one for each counterparty. For more info on the uses for
-            // this, see https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2018-November/016518.html
-            if (nSize >  EXTRA_DESCENDANT_TX_SIZE_LIMIT ||
-                    !pool.CalculateMemPoolAncestors(entry, setAncestors, 2, nLimitAncestorSize, nLimitDescendants + 1, nLimitDescendantSize + EXTRA_DESCENDANT_TX_SIZE_LIMIT, dummy_err_string)) {
-                return state.Invalid(ValidationInvalidReason::TX_MEMPOOL_POLICY, false, REJECT_NONSTANDARD, "too-long-mempool-chain", errString);
+            if (setConflicts.size() == 1) {
+                // In general, when we receive an RBF transaction with mempool conflicts, we want to know whether we
+                // would meet the chain limits after the conflicts have been removed. However, there isn't a practical
+                // way to do this short of calculating the ancestor and descendant sets with an overlay cache of
+                // changed mempool entries. Due to both implementation and runtime complexity concerns, this isn't
+                // very realistic, either, thus we we try to ensure a limited set of transaction is RBF'able despite
+                // mempool conflicts here. Importantly, we want to ensure that transactions which were accepted using
+                // the below carve-out are able to be RBF'ed, without impacting the security the carve-out provides
```

Comment:
you should probably label the thing you're calling `carve-out` for those not reading git history.